src/freezing/freezing_main.py

```diff
@@ -81,10 +82,10 @@ def run_freezing(
     #     )
 
     # # Read the freezing files from the last run and apply them
-    # constructed_df = apply_freezing(
-    #     main_snapshot, config, check_file_exists, read_csv, write_csv, run_id
-    # )
-    # constructed_df.reset_index(drop=True, inplace=True)
+    constructed_df = apply_freezing(
```

Comment:
This just needs to run if run_updates_and_freeze = True

Reply:
Can you update when it runs, and the input:
    # Read the freezing files from the last run and apply them
    if run_updates_and_freeze:
        frozen_data = read_frozen_csv(config, read_csv)
        prepared_frozen_data = apply_freezing(
            frozen_data, config, check_file_exists, read_csv, run_id, FreezingLogger
        )
        prepared_frozen_data.reset_index(drop=True, inplace=True)